PopcornEditor/src/butter.js

```diff
@@ -922,67 +922,127 @@ window.Butter = {
+                else if ((mat = initialMediaSource.match(/\/t=([\d\.]+)$/))){
+                  post += '?start='+mat[1];
+                  initialMediaSource = initialMediaSource.replace(/\/t=([\d\.]+)$/,'');
+                }
+
+                // now expand the "short" source version to the full version
+                initialMediaSource = 'https://archive.org/details/' + initialMediaSource + post;
+              }
+
+              _logger.log('ADDING initialMediaSource: '+initialMediaSource);
+              initialMediaList.push(initialMediaSource);
+            });
+          }
+
+          return initialMediaList;
+        }
+
         function loadConfigDefault() {
           // if previous attempt failed,
           // try loading data from the savedDataUrl value in the config
           loadFromSavedDataUrl( _config.value( "savedDataUrl" ), function( savedData ) {
-            var initialMediaSource;
+            var trackTime=0;
+
+            var initialMediaList = initialMediaSources( qs );
 
-            if ( !qs.initialMedia ) {
+            if ( !initialMediaList.length ) {
               return projectDataReady( savedData );
             }
 
-            initialMediaSource = decodeURIComponent( qs.initialMedia );
-
-            // If we successfully retrieve data for that initial media we will hand write it
-            // into the default project data before the import.
-            MediaUtil.getMetaData( initialMediaSource, function onSuccess( data ) {
-              var media = savedData.media[ 0 ],
-                  track;
-
-              if ( media ) {
-                media.url = "#t=," + data.duration;
-                media.duration = data.duration;
-
-                track = media.tracks[ 0 ];
-
-                if ( track ) {
-                  track.trackEvents.push({
-                    id: "TrackEvent1",
-                    type: "sequencer",
-                    popcornOptions: {
-                      source: [ data.source ],
-                      start: 0,
-                      end: data.duration,
-                      title: data.title,
-                      from: data.from,
-                      duration: data.duration,
-                      target: "video-container",
-                      fallback: "",
-                      zindex: 1000,
-                      id: "TrackEvent1",
-                      linkback: data.linkback,
-                      thumbnailSrc: data.thumbnail
-                    }
-                  });
+            var media = savedData.media[ 0 ];
+            if (media)
+              media.duration = 0;
+
+            var num2get = initialMediaList.length;
+            $(initialMediaList).each(function(idx,initialMediaSource){
+              // If we successfully retrieve data for that initial media we will hand write it
+              // into the default project data before the import.
+              // xxxx 2+ media is race condition adding now!
```

Comment:
[reads more code...]
actually, looks like that more or less *is* what the existing code and this branch are doing -- effectively adding media to a "saved/stub project" before import.  ok, phew! so that seems indeed like best practices, so will do brion's idea now..